src/cache/precipitation_precomputed_quantities.jl

```diff
@@ -57,5 +79,286 @@ function set_sedimentation_precomputed_quantities!(Y, p, t)
+        ᶜSᵖ,
+        ᶜSᵖ_snow,
+        ᶜSqₗᵖ,
+        ᶜSqᵢᵖ,
+        ᶜSqᵣᵖ,
+        ᶜSqₛᵖ,
+        Y.c.ρ,
+        q_rai,
+        q_sno,
+        ᶜts,
+        dt,
+        cmp,
+        thp,
+    )
+
+    # compute precipitation sinks
+    # (For now only done on the grid mean)
+    compute_precipitation_sinks!(
+        ᶜSᵖ,
+        ᶜSqᵣᵖ,
+        ᶜSqₛᵖ,
+        Y.c.ρ,
+        q_rai,
+        q_sno,
+        ᶜts,
+        dt,
+        cmp,
+        thp,
+    )
+    return nothing
+end
+function set_precipitation_cache!(
+    Y,
+    p,
+    ::Microphysics1Moment,
+    ::Union{DiagnosticEDMFX, PrognosticEDMFX},
+)
+    error("Not implemented yet")
+
+    #FT = Spaces.undertype(axes(Y.c))
+    #(; dt) = p
+    #(; ᶜts, ᶜqᵣ, ᶜqₛ, ᶜwᵣ, ᶜwₛ, ᶜu) = p.precomputed
+    #(; ᶜΦ) = p.core
+    ## Grid mean precipitation sinks
+    #(; ᶜSqₜᵖ, ᶜSqᵣᵖ, ᶜSqₛᵖ, ᶜSeₜᵖ) = p.precipitation
+    ## additional scratch storage
+    #ᶜSᵖ = p.scratch.ᶜtemp_scalar
+    #ᶜ∇T = p.scratch.ᶜtemp_CT123
+
+    ## get thermodynamics and 1-moment microphysics params
+    #(; params) = p
+    #cmp = CAP.microphysics_1m_params(params)
+    #thp = CAP.thermodynamics_params(params)
+
+    ## zero out the helper source terms
+    #@. ᶜSqₜᵖ = FT(0)
+    #@. ᶜSqᵣᵖ = FT(0)
+    #@. ᶜSqₛᵖ = FT(0)
+    #@. ᶜSeₜᵖ = FT(0)
+    ## compute precipitation sinks
+    ## (For now only done on the grid mean)
+    #compute_precipitation_sinks!(
+    #    ᶜSᵖ,
+    #    ᶜSqₜᵖ,
+    #    ᶜSqᵣᵖ,
+    #    ᶜSqₛᵖ,
+    #    ᶜSeₜᵖ,
+    #    Y.c.ρ,
+    #    ᶜqᵣ,
+    #    ᶜqₛ,
+    #    ᶜts,
+    #    ᶜΦ,
+    #    dt,
+    #    cmp,
+    #    thp,
+    #)
+    return nothing
+end
+
+"""
+    set_precipitation_surface_fluxes!(Y, p, precipitation model)
+
+Computes the flux of rain and snow at the surface. For the 0-moment microphysics
+it is an integral of the source terms in the column. For 1-moment microphysics
+it is the flux through the bottom cell face.
+"""
+set_precipitation_surface_fluxes!(Y, p, _) = nothing
+function set_precipitation_surface_fluxes!(
+    Y,
+    p,
+    precip_model::Microphysics0Moment,
+)
+    ᶜT = p.scratch.ᶜtemp_scalar
+    (; ᶜts) = p.precomputed  # assume ᶜts has been updated
+    (; ᶜ3d_rain, ᶜ3d_snow, ᶜS_ρq_tot, ᶜS_ρe_tot) = p.precomputed
+    (; surface_rain_flux, surface_snow_flux) = p.precomputed
+    (; col_integrated_precip_energy_tendency) = p.conservation_check
+
+    # update total column energy source for surface energy balance
+    Operators.column_integral_definite!(
+        col_integrated_precip_energy_tendency,
+        ᶜS_ρe_tot,
+    )
+    # update surface precipitation fluxes in cache for coupler's use
+    thermo_params = CAP.thermodynamics_params(p.params)
+    T_freeze = TD.Parameters.T_freeze(thermo_params)
+    FT = eltype(p.params)
+    @. ᶜT = TD.air_temperature(thermo_params, ᶜts)
+    @. ᶜ3d_rain = ifelse(ᶜT >= T_freeze, ᶜS_ρq_tot, FT(0))
+    @. ᶜ3d_snow = ifelse(ᶜT < T_freeze, ᶜS_ρq_tot, FT(0))
```

Comment:
```suggestion
    ᶜ3d_rain = @. lazy(ifelse(ᶜT >= T_freeze, ᶜS_ρq_tot, FT(0)))
    ᶜ3d_snow @. lazy(ifelse(ᶜT < T_freeze, ᶜS_ρq_tot, FT(0)))
```
I think this might work, in which case, you could eliminate this variable from the cache entirely!

Reply:
I think it did! Thank you